classes/class3/cat-app/routes/cats.js

```diff
@@ -0,0 +1,60 @@
+var mongoose = require('mongoose');
+
+var names = ['Kurumi', 'Matcha', 'Peppermint', 'Botchan', 'Jessie', 'Charcoal', 'Alex'];
+var colors = ['tan', 'black', 'white', 'orange', 'grey', 'chocolate'];
+
+var cat_schema = mongoose.Schema({
+	name: String,
+	age: Number,
+	color: String },
+	{collection: 'cats'}
+);
+
+var Cat = mongoose.model('Cat', cat_schema);
+
+var list = function(req, res) {
+	Cat.find({}, function(err, list) { 
+		if (err) console.log('Cannot List Cats', err);
+		res.render('cats', {'cats': list})
+	});
+};
+
+var create = function(req, res) {
+	var cat = new Cat({name: names[Math.floor(Math.random()*names.length)],
+		age: Math.floor(Math.random()*25),
+		color: colors[Math.floor(Math.random()*colors.length)]});
+	
+	cat.save(function (err) {
+		if (err) console.log('Problem Creating Kitten', err);
+	})
+	res.render('new', {'cat': cat});
+};
+
+var colorSort = function(req, res) {
+	var address = req.url.split('/');
+	var col = address[address.length-1];
```

Comment:
Here's where `/:color` would have been helpful. Instead of these two lines parsing the color value, you can just use `req.params.color`.

Also, there's no need to abbreviate variable names quite so much. `color` would be a fine name that helps convey the value it contains, more so than `col`.